### PR TITLE
Handle embedded tweets in substack emails

### PR DESCRIPTION
**Pull request by jacksonh**
- Handle embedded tweets in substack emails
- Add CSS to handle static tweets to frontend

### CODE REVIEW

File: packages/content-handler/test/newsletter.test.ts [hunk 1]
```diff
@@ -129,6 +130,41 @@ describe('Newsletter email test', () => {
...
+      const handler = await getNewsletterHandler({
+        html,
+        postHeader: '',
+        from: '',
+        unSubHeader: '',
+      })
+      expect(handler).to.be.instanceOf(SubstackHandler)
+
+      const dom = parseHTML(html).document
+      expect(handler?.shouldPreParse(url, dom)).to.be
```
Comment (sywhb) [inline packages/content-handler/test/newsletter.test.ts:160]: I guess we left out the expected result here

File: packages/readabilityjs/Readability.js [hunk 2]
```diff
@@ -2774,17 +2784,30 @@ Readability.prototype = {
...
       }
       return false;
     });
   },
 
+  _isOmnivoreNode: function(node) {
+    var walk = node
+
+    while (walk) {
+      if (walk.className && node.className.startsWith("_omnivore")) {
```
Comment (sywhb) [inline packages/readabilityjs/Readability.js:2803]: Is the `node` here supposed to be `walk`?

**Review by sywhb — COMMENTED** [review]
Nice!

File: packages/content-handler/test/newsletter.test.ts (hunk 1 above)
Comment (jacksonh) [reply packages/content-handler/test/newsletter.test.ts:160]: I think this verifies that its true, but I'll make it more clear.

File: packages/readabilityjs/Readability.js (hunk 2 above)
Comment (jacksonh) [reply packages/readabilityjs/Readability.js:2803]: 😬  yes!

File: packages/content-handler/src/newsletters/substack-handler.ts [hunk 3]
```diff
@@ -85,4 +95,46 @@ export class SubstackHandler extends ContentHandler {
...
+      recurse(tweet, (n: Node) => {
+        if (isHTMLElement(n)) {
+          const className = n.className
+          if (className.startsWith('tweet-')) {
+            n.className = preClassName + className
+          }
+          n.removeAttribute('style')
+        }
+      })
+    }
```
[the remark this answers is not in the document]
Comment (jacksonh) [reply packages/content-handler/src/newsletters/substack-handler.ts]: :nice:

**Comment by jacksonh** [comment]
OK, i think this looks better now. Thanks for the feedback.

File: packages/api/test/utils/parser.test.ts [hunk 4]
```diff
@@ -38,11 +38,17 @@ describe('parseMetadata', async () => {
 
 describe('parsePreparedContent', async () => {
   it('gets published date when JSONLD fails to load', async () => {
+    nock('https://stratechery.com:443', {"encodedQueryParams":true})
+      .get('/wp-json/oembed/1.0/embed')
+      .query({"url":"https%3A%2F%2Fstratechery.com%2F2016%2Fits-a-tesla%2F"})
+      .reply(401)
+
```
Comment (sywhb) [inline packages/api/test/utils/parser.test.ts:45]: 👍

**Review by sywhb — APPROVED** [review]
LGTM!

**Pull request merged**